eqcorrscan/utils/correlate.py

```diff
@@ -598,8 +598,10 @@ def fftw_multi_normxcorr(template_array, stream_array, pad_array, seed_ids):
     n_templates = template_array[seed_ids[0]].shape[0]
     image_len = stream_array[seed_ids[0]].shape[0]
     fft_len = next_fast_len(template_len + image_len - 1)
-    template_array = np.array(list(template_array.values())).flatten(order='C')
-    stream_array = np.array(list(stream_array.values())).flatten(order='C')
+    template_list = [template_array[x] for x in seed_ids]
+    stream_list = [stream_array[x] for x in seed_ids]
+    template_array = np.array(template_list).flatten(order='C')
+    stream_array = np.array(stream_list).flatten(order='C')
```

Comment:
This looks like it fixed those fails, nice catch - the failing tests look to me to all be network timeout related :-1: so I will rerun those until they don't crap out.
@chrisdjscott has started to move some of the reshaping into the C-func #138 to make this faster, and this may well clash with his thing, but I can work it out later on.  Looking at #138 though I am surprised that we haven't run into this issue there?

Reply:
@calum-chamberlain I should probably make this change in my branch too - it should be easy to merge later as only a couple of lines.

Reply:
Sounds good @chrisdjscott. I'm still not sure why this didn't emerge earlier in the testing from updating the internal correlation routines and am a little concerned. @d-chambers do you know if this fail happened thanks to a change in tests, or was it just from changing `correlate.py`?

Reply:
I have no idea but there clearly was an order dependence on un-ordered dictionaries. I have a feeling we would have to dive very deep into the C-python implementation to understand how it didn't affect the code before, but it is possible that it might have caused some subtle bugs that didn't fail the tests, who knows?

Reply:
Agreed on the dependence on the order - I hadn't even considered it because it never presented itself as an issue. I wonder if the order was preserved inside the function before, which has now been lost by genertaing the dictionaries in another funcion (which btw is IMHO much cleaner and better 👍 )? - either way this is the *right* way to do this and I should have done it originally. 

I'm just debating whether I patch master with this? I'm finding it hard to find a justification for why it used to work, which is pushing me towards patching. Open to opinions @cjhopp and @d-chambers ?

Reply:
Patching is not a bad idea unless you feel we are close to another version release.

Reply:
If I'm going to patch it then it would be another release - it would be nice to include this in the next release, it's feeling close to me, what do you guys think? @cjhopp @d-chambers 
If it is close (next week or so) then I'd be tempted to wait, mostly because I can't reproduce the issue with the old/current code...